packages/kilt-did/README.md

```diff
@@ -3,6 +3,320 @@
+
+```typescript
+import Keyring from '@polkadot/keyring'
+
+import Kilt from '@kiltprotocol/sdk-js'
+import { BlockchainUtils } from '@kiltprotocol/chain-helpers'
+import {
+  DefaultResolver,
+  DemoKeystore,
+  DidUtils,
+  SigningAlgorithms
+} from '@kiltprotocol/did'
+import { KeyRelationship } from '@kiltprotocol/types'
+
+// Initialise connection to a KILT blockchain node.
+await Kilt.init({ address: 'ws://127.0.0.1:9944' })
+
+// Generate the KILT account that will submit the DID creation tx to the KILT blockchain.
+// It must have enough funds to pay for the tx execution fees.
+const aliceKiltAccount = new Keyring({
+  type: 'ed25519',
+  // KILT has registered the ss58 prefix 38
+  ss58Format: 38,
+}).createFromUri('//Alice')
+
+// Instantiate the demo keystore.
+const keystore = new DemoKeystore()
+
+// Generate seed for the authentication key.
+const authenticationSeed = '0x123456789'
+
+// Ask the keystore to generate a new keypar to use for authentication.
+const authenticationKeyPublicDetails = await keystore.generateKeypair({
+  seed: authenticationSeed,
+  alg: SigningAlgorithms.Ed25519,
+})
+
+// Generate the DID-signed creation extrinsic.
+// The extrinsic is unsigned and contains the DID creation operation signed with the DID authentication key.
+const { submittable, did } = await DidUtils.writeDidfromPublicKeys(keystore, {
+  [KeyRelationship.authentication]: {
+    publicKey: authenticationKeyPublicDetails.publicKey,
+    type: authenticationKeyPublicDetails.alg,
+  },
+})
+// Will print `did:kilt:4sxSYXakw1ZXBymzT9t3Yw91mUaqKST5bFUEjGEpvkTuckar`.
+console.log(did)
+
+// Submit the DID creation tx to the KILT blockchain after signing it with the KILT account specified.
+await BlockchainUtils.signAndSubmitTx(submittable, aliceKiltAccount, {
+  resolveOn: BlockchainUtils.IS_IN_BLOCK,
+})
+
+const fullDid = (await DefaultResolver.resolveDoc(did)) as FullDidDetails
+```
+
+If additional keys and external services are to be specified, then they can be included in the DID create operation.
+
+```typescript
+await Kilt.init({ address: 'ws://127.0.0.1:9944' })
+
+const aliceKiltAccount = new Keyring({
+  type: 'ed25519',
+  ss58Format: 38,
+}).createFromUri('//Alice')
+
+const keystore = new DemoKeystore()
+
+const authenticationSeed = '0x123456789'
+
+const authenticationKeyPublicDetails = await keystore.generateKeypair({
+  seed: authenticationSeed,
+  alg: SigningAlgorithms.Ed25519,
+})
+
+// Generate seed for the encryption key.
+const encryptionSeed = '0x987654321'
+
+// Ask the keystore to generate a new keypar to use for encryption.
+const encryptionKeyPublicDetails = await keystore.generateKeypair({
+  seed: encryptionSeed,
+  alg: EncryptionAlgorithms.NaclBox,
+})
+
+// Define the set of URLs pointing to the external services exposed by the DID subject.
+// All URLs must point to the same resource, which must match the provided hash.
```

Comment:
Does our code actually verify that the resource is the same and matches the hash? As I was trying to read the requirements, I was really confused where they come from? My guess is they are a small part of "how to make a valid DID" knowledge. Should we rather refer to some document on that topic, and skip this detail here? To me it looks unnecessary and confusing in this place.

Reply:
That part, relative to the services resolver, is not yet implemented in the SDK, and will be the next big chunk of work for our next sprint. Nevertheless, I agree that is probably too much information to put in here 👍